web/handlers.go

```diff
@@ -118,21 +119,20 @@ func generateDevCSP(c Context) string {
 			// Honour only supported keys
 			switch devFlagKey {
 			case "unsafe-eval", "unsafe-inline":
-				devCSPMap[devFlagKey] = true
+				devCSP = append(devCSP, "'"+devFlagKey+"'")
```

Comment:
Wouldn't this repeat `unsafe-eval` and `unsafe-inline` again if they are set? It's not super clear to me what is the benefit here by changing map to slice.

Reply:
Yeah, it would. I don't imagine repeating things will come up particularly often though, and I don't think it should cause any issues to repeat stuff in that header.

Changing it to a slice was because I was having a bit of trouble reading the way the code was written before because of how it filtered the valid fields twice (which I had to remove since the new ones are arbitrary). It also made it so that I could just `strings.Join` them together at the end to turn it into a string instead of having to loop through a map with an undefined order.

Reply:
I think the map was fine, but `for _, devCSPFlag := range supportedCSPFlags {` wasn't necessary. There's no need to loop through the supported flags again.

I am wondering if we can keep the map, set the keys as usual, and then just iterate the keys once to create a slice. It's okay to let the order be undefined. Or we can `sort.Slice` after that if you prefer.

That way the items aren't repeated, and there's no filtering twice as well. Let me know your thoughts.

Reply:
Having the order be undefined would probably make the unit tests flaky if the order changes run-to-run, and sorting it feels a little weird to me. What if we just wrap these two with a check to `slice.Contains` since this is the only place duplicates might be added?

Reply:
That works :+1: